src/webgpu/api/validation/encoding/cmds/compute_pass.spec.ts

```diff
@@ -0,0 +1,128 @@
+    if (state === 'valid') {
+      return this.createNoOpComputePipeline();
+    }
+
+    return this.createErrorComputePipeline();
+  }
+
+  createIndirectBuffer(state: 'valid' | 'invalid' | 'destroyed', data: Uint32Array): GPUBuffer {
+    const descriptor: GPUBufferDescriptor = {
+      size: data.byteLength,
+      usage: GPUBufferUsage.INDIRECT | GPUBufferUsage.COPY_DST,
+    };
+
+    if (state === 'invalid') {
+      descriptor.usage = 0xffff; // Invalid GPUBufferUsage
+    }
+
+    this.device.pushErrorScope('validation');
+    const buffer = this.device.createBuffer(descriptor);
+    this.device.popErrorScope();
+
+    if (state === 'valid') {
+      this.queue.writeBuffer(buffer, 0, data);
+    }
+
+    if (state === 'destroyed') {
+      buffer.destroy();
+    }
+
+    return buffer;
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('set_pipeline')
+  .desc(
+    `- Tests using compute pipelines
+  - An error should be generated when using an 'invalid' pipeline`
+  )
+  .params(poptions('state', ['valid', 'invalid'] as const))
+  .fn(t => {
+    const pipeline = t.createComputePipeline(t.params.state);
+    const { encoder, finish } = t.createEncoder('compute pass');
+    encoder.setPipeline(pipeline);
+    t.expectValidationError(() => {
+      finish();
+    }, t.params.state === 'invalid');
+  });
+
+g.test('dispatch_sizes')
+  .desc(
+    `- For both 'direct' and 'indirect' dispatch
+  - Tests using the following workgroup sizes: {[0, 0, 0], [1, 1, 1]}
+  - An error should occur when the number exceeds <fill number here>`
+  )
+  .params(
+    params()
+      .combine(poptions('dispatchType', ['direct', 'indirect'] as const))
+      .combine(
+        poptions('workSizes', [
+          [0, 0, 0],
+          [1, 1, 1],
+          // TODO: Add tests for workSizes right under and above upper limit once the limit has been decided.
+        ] as [number, number, number][])
+      )
+  )
+  .fn(t => {
+    const pipeline = t.createNoOpComputePipeline();
+    const [x, y, z] = t.params.workSizes;
+    const { encoder, finish } = t.createEncoder('compute pass');
+    encoder.setPipeline(pipeline);
+    if (t.params.dispatchType === 'direct') {
+      encoder.dispatch(x, y, z);
+    } else if (t.params.dispatchType === 'indirect') {
+      encoder.dispatchIndirect(t.createIndirectBuffer('valid', new Uint32Array([x, y, z])), 0);
+    }
+    t.queue.submit([finish()]);
+  });
+
+const kBufferData = new Uint32Array(6).fill(1);
+g.test('indirect_dispatch_buffer')
+  .desc(
+    `- For 'indirect' dispatch
+  - Tests that indirect buffers:
+    - 'invalid', 'destroyed' generate an error
+    - 'valid' buffers do not generate an error
+  - Tests that, for a buffer with 6 elements, indirect offsets:
+    - 0, 'sizeof(uint32)', and '3 * sizeof(uint32)' do not generate an error
+    - An error should be generate by the following:
+      - 1: non-multiple of 4
+      - '4 * sizeof(uint32): z-components outside of buffer`
```

Comment:
Here's this plan formatted the way I had in mind - note it's a little lighter on case details since they're right below. WDYT? (I'm hoping to make all of the plans more uniform in the future.)

```suggestion
    `
Test dispatchIndirect validation by submitting various dispatches with a no-op
pipeline and an indirectBuffer with 6 elements.
- indirectBuffer: {'valid', 'invalid', 'destroyed'}
- indirectOffset:
  - valid, within the buffer: {beginning, middle, end} of the buffer
  - invalid, non-multiple of 4
  - invalid, the last element is outside the buffer
```

Reply:
Looks good to me. Looks like you added a line break at 80 col. Was that intentional? If so, should we enforce 80 cols with ESLint?

BTW, I used the demo as an example for the "style". https://github.com/gpuweb/cts/blob/5ab9cd41d491881e3a40575a3841c27fc0ea4a8a/src/demo/a/b/c.spec.ts#L10-L14
We should probably make that the demo match the correct styling.

Reply:
> Looks good to me. Looks like you added a line break at 80 col. Was that intentional? If so, should we enforce 80 cols with ESLint?

I added it in an arbitrary place (can't see the columns since I was editing in GitHub), but we do enforce a column limit (I think it's 100 cols). The problem is this is inside a string so idk if ESLint can wrap it for us (I guess it would insert `\\\n`).

> We should probably make that the demo match the correct styling.

Oh, very excellent point. I didn't think very deeply about the formatting when I wrote that (was just trying to make sure the text showed up at all on /standalone/). Would you mind making that tweak, and make sure there's an example in `examples.spec.ts` too? (can be the same)

Reply:
I just noticed `printWidth: 100` on prettier.config.js. It doesn't seem to work on some lines, 
https://github.com/gpuweb/cts/blob/c66e2bd6f5f60b45c5acdca197110910f4e13f33/src/webgpu/api/validation/encoding/cmds/compute_pass.spec.ts#L132
It seems like prettier is forcing that statement to be in a single line. Adding `max-len` to eslint just makes it error out with no way to fix it.

Also, I'll update the examples.

Reply:
That's weird, I don't think I have run into cases where prettier fails like that. Thanks for looking into this.